### PR TITLE
Modify comment for hasImplicitTableAlias() override

### PR DESCRIPTION
**Pull request by ljfgem**
## Summary
This PR adds details to `hasImplicitTableAlias()` override comment.

## Test
No test needed.

### CODE REVIEW

File: coral-hive/src/main/java/com/linkedin/coral/transformers/CoralRelToSqlNodeConverter.java [hunk 1]
```diff
@@ -92,8 +92,24 @@ public boolean requireCastOnString() {
...
+       * For the following input SQL:
+       * CREATE TABLE default.complex(s struct(name:string, age:int))
+       * SELECT tbl.str.a FROM db.tbl
+       *
+       * With this override, the converted SqlNode is:
+       * SELECT `complex`.`s`.`name` AS `name`
+       * FROM `default`.`complex` AS `complex`
+       *
+       * Without this override, the converted SqlNode is:
+       * SELECT `s`.`name` AS `name`
```
Comment (wmoustafa) [inline coral-hive/src/main/java/com/linkedin/coral/transformers/CoralRelToSqlNodeConverter.java:107]: Why does the alias show up here in the case of "without the override"?

Comment (ljfgem) [reply coral-hive/src/main/java/com/linkedin/coral/transformers/CoralRelToSqlNodeConverter.java:107]: This field alias is not related to table alias generated by this override, this override won't affect the field alias like `name`

**Review by aastha25 — COMMENTED** [review]
Thanks for the Pr @ljfgem

**Comment by ljfgem** [comment]
@wmoustafa @aastha25 Thanks for your detailed reviews! I have solved all the comments, PTAL.

File: coral-hive/src/main/java/com/linkedin/coral/transformers/CoralRelToSqlNodeConverter.java [hunk 2]
```diff
@@ -92,8 +92,36 @@ public boolean requireCastOnString() {
...
+       * WHERE `complex`.`s`.`age` = 25
+       *
+       * Without this override, the converted SqlNode is:
+       * SELECT `split`(`s`.`name`, ' ') AS `name_array`
+       * FROM `default`.`complex`
+       * WHERE `s`.`age` = 25
+       *
+       * Without this override, if we want to get the data type of a struct field like `s`.`name`, validation will fail
+       * because Calcite uses {@link org.apache.calcite.rel.type.StructKind#FULLY_QUALIFIED} for standard SQL and it
+       * expects each struct field to be referenced explicitly with table alias in the method
```
Comment (aastha25) [inline coral-hive/src/main/java/com/linkedin/coral/transformers/CoralRelToSqlNodeConverter.java:121]: could you rephrase "each struct field" to emphasize "each field inside the struct"? 
Iirc, the struct field, `s`, is not an issue. the field inside the struct, `s.name` fails data type derivation unless fully qualified. Pl correct if this is not the case

File: coral-hive/src/main/java/com/linkedin/coral/transformers/CoralRelToSqlNodeConverter.java [hunk 3]
```diff
@@ -92,8 +92,36 @@ public boolean requireCastOnString() {
...
+       * WHERE complex.s.age = 25;
+       *
+       * The input RelNode is:
+       * LogicalProject(name_array=[split($0.name, ' ')])
+       *   LogicalFilter(condition=[=($0.age, 25)])
+       *     LogicalTableScan(table=[[hive, default, complex]])
+       *
+       * With this override, the converted SqlNode is:
+       * SELECT `split`(`complex`.`s`.`name`, ' ') AS `name_array`
+       * FROM `default`.`complex` AS `complex`
```
Comment (aastha25) [inline coral-hive/src/main/java/com/linkedin/coral/transformers/CoralRelToSqlNodeConverter.java:111]: I'm also just realizing that the javadoc doesn't justify the change `FROM `default`.`complex`` -> `FROM `default`.`complex` AS `complex`

Could you please add a 1 liner to elaborate why & how the `AS` clause was added here?

Comment (ljfgem) [reply coral-hive/src/main/java/com/linkedin/coral/transformers/CoralRelToSqlNodeConverter.java:111]: Thanks for suggestion, added in the end, it's just a side effect of this override.

File: coral-hive/src/main/java/com/linkedin/coral/transformers/CoralRelToSqlNodeConverter.java [hunk 4]
```diff
@@ -92,8 +92,36 @@ public boolean requireCastOnString() {
...
+       * Without this override, the converted SqlNode is:
+       * SELECT `split`(`s`.`name`, ' ') AS `name_array`
+       * FROM `default`.`complex`
+       * WHERE `s`.`age` = 25
+       *
+       * Without this override, if we want to get the data type of a struct field like `s`.`name`, validation will fail
+       * because Calcite uses {@link org.apache.calcite.rel.type.StructKind#FULLY_QUALIFIED} for standard SQL and it
+       * expects each struct field to be referenced explicitly with table alias in the method
+       * {@link org.apache.calcite.sql.validate.DelegatingScope#fullyQualify(SqlIdentifier)}.
+       * Therefore, we need to generate `complex`.`s`.`name` with this override, which will also affect other non-struct
```
[the remark this answers is not in the document]
Comment (ljfgem) [reply coral-hive/src/main/java/com/linkedin/coral/transformers/CoralRelToSqlNodeConverter.java:123]: Yes

**Review by ljfgem — COMMENTED** [review]
@aastha25 Thanks for your review, modified the PR based on your comment. PTAL.

File: coral-hive/src/main/java/com/linkedin/coral/transformers/CoralRelToSqlNodeConverter.java [hunk 5]
```diff
@@ -92,8 +92,36 @@ public boolean requireCastOnString() {
       }
 
       /**
-       * Override this method so that there will be explicit and correct table alias for selected fields, which is necessary for
-       * data type derivation on SqlNodes
+       * Override this method so that there will be explicit table alias for all the fields, which is necessary for
```
Comment (wmoustafa) [inline coral-hive/src/main/java/com/linkedin/coral/transformers/CoralRelToSqlNodeConverter.java:95]: `so that table alias is prepended to all field references (e.g., "table.column" or "table.struct.filed" instead of "column" or "struct.field")`.

File: coral-hive/src/main/java/com/linkedin/coral/transformers/CoralRelToSqlNodeConverter.java [hunk 6]
```diff
@@ -92,8 +92,36 @@ public boolean requireCastOnString() {
...
+       * SELECT split(complex.s.name, ' ') name_array
+       * FROM default.complex
+       * WHERE complex.s.age = 25;
+       *
+       * The input RelNode is:
+       * LogicalProject(name_array=[split($0.name, ' ')])
+       *   LogicalFilter(condition=[=($0.age, 25)])
+       *     LogicalTableScan(table=[[hive, default, complex]])
+       *
+       * With this override, the converted SqlNode is:
```
Comment (wmoustafa) [inline coral-hive/src/main/java/com/linkedin/coral/transformers/CoralRelToSqlNodeConverter.java:109]: converted --> generated

File: coral-trino/src/main/java/com/linkedin/coral/trino/rel2trino/TrinoSqlDialect.java [hunk 7]
```diff
@@ -31,8 +31,36 @@ public boolean supportsCharSet() {
   }
 
   /**
-   * Override this method so that there will be explicit and correct table alias for selected fields, which is necessary for
-   * data type derivation on SqlNodes
+   * Override this method so that there will be explicit table alias for all the fields, which is necessary for
+   * data type derivation on struct fields.
+   *
+   * For the following input SQL:
+   * CREATE TABLE default.complex(s struct(name:string, age:int));
```
Comment (wmoustafa) [inline coral-trino/src/main/java/com/linkedin/coral/trino/rel2trino/TrinoSqlDialect.java:38]: This comment and the override itself are going away in the future, correct?

Comment (ljfgem) [reply coral-trino/src/main/java/com/linkedin/coral/trino/rel2trino/TrinoSqlDialect.java]: Yes

**Comment by ljfgem** [comment]
@wmoustafa Modified the comment, PTAL.
